source/grid.js

```diff
@@ -153,22 +172,43 @@ this.addToGridFromWorld = function( object, worldCoord ) {
 }
 
 this.removeFromGrid = function( nodeID, gridCoord ) {
-    var objects = this.getObjectsAtCoord( gridCoord );
-    var index = objects.indexOf( nodeID );
-    if ( index !== -1 ) {
-        for ( var i = index; i < objects.length - 1; i++ ) {
-            objects[ i ] = objects[ i + 1 ];
+
+    var node = this.scene.findByID( this.scene, nodeID ); 
+    var bArea = node.boundingAreaSize; 
+
+    for( var x = 0; x < bArea[ 0 ]; x++ ){
+        for( var y = 0; y < bArea[ 1 ]; y++ ){
+            var currTileCoord = [ gridCoord[ 0 ] + x , gridCoord[ 1 ] - y ]; 
+            var objects = this.getObjectsAtCoord( currTileCoord );
+            var index = objects.indexOf( nodeID );
+
+            if ( index !== -1 ) {
+                for ( var i = index; i < objects.length - 1; i++ ) {
+                    objects[ i ] = objects[ i + 1 ];
+                }
+                objects.length--;
+            } else {
+                this.logger.errorx( "removeFromGrid", "Unable to find tile with " +
+                    "ID: " + nodeID );
+                return false;
+            }
```

Comment:
I did test with multiple turns, but I'll go back and test it again, this time documenting exactly what I did to test it.

Right now speed can't be too bad since our heavy rover is 2x2. But it's an O(n^2) check, assuming the rovers are square. Is getObjectsAtCoord() something that might be slow, as well? Ultimately it does a lookup into this.tiles, and I remember Kevin saying that calls into "this" (assuming "this" is a vwf object) are expensive, since they need to be synchronized. 

One easy way to speed things up would be to consider the direction that the rover is moving in, and then only check "front" tiles, because those are the only ones that are going to collide with something else.